BUNDLE_FORMAT.md

```diff
@@ -58,3 +58,92 @@ all the dependencies to the current environment.
 When sourcing this `setup.sh` it expects `COLCON_BUNDLE_PREFIX` to
 be set to the folder where the contents of `bundle.tar` have 
 been extracted.
+
+# Bundle Format - V2
+
+## Summary
+
+The output of the bundle verb is a tar file containing multiple sub-files
+
+1. version - A file with the version of the bundle file format
+1. metadata.tar - A tar archive containing various metadata
+1. N *.tar files - These tar archives contain overlay'able contents
+
+** Overlay'able - The top level of each of the archives contains a setup.sh
+that can be sources to overlay this archive's contents on top of the 
+current environment
+
+### version
+
+Contains a single version number formatted \<major version\>.\<minor version\>. The minor version
+will be incremented whenever a backwards compatible change is made. The major version will
+be incremented if there is a non-backwards compatible change to the bundle format.
+
+### metadata.tar.gz
+
+All files contained within metadata.tar are formatted as a JSON dictionary. This allows
+for backwards compatible additions to any of the files. Order is not guaranteed for this
+archive.
+
+**installers.json:**
+
+    {
+	  "apt": {
+		"installed_packages": [
+		  {
+			"version": "2.26.1",
+			"name": "pacakge1"
+		  },
+		  {
+			"version": "1.0.6-8",
+			"name": "package2"
+		  }
+		]
+	  }
+	  "pip3": {
+		"installed_packages": [
+		  {
+			"version": "1.11.13",
+			"name": "package1"
+		  },
+		  {
+			"version": "1.8.3",
+			"name": "package2"
+		  }
+		]
+	  }
+    }
+    
+**overlays.json:**
```

Comment:
Should we add details so that this is not confused with overlayed/union filesystem.

Reply:
Is there a better name? This is just the first thing that comes to my mind since that's what verbage OSRF uses.

http://wiki.ros.org/catkin/Tutorials/workspace_overlaying

Reply:
I don't have a better alternative either. However having documentation would help disambiguate to good extend.

Reply:
Added some more documetnation and the link I shared above to help disambiguate what we are trying to do.